build-tools/packages/build-cli/src/library/changesets.ts

```diff
@@ -3,86 +3,262 @@
  * Licensed under the MIT License.
  */
 
+import { readFile } from "node:fs/promises";
 import path from "node:path";
 import { VersionBumpType } from "@fluid-tools/version-tools";
-import { Logger } from "@fluidframework/build-tools";
+import { Logger, type ReleaseNotesSectionName } from "@fluidframework/build-tools";
 import { compareAsc, formatISO, parseISO } from "date-fns";
 import globby from "globby";
 import matter from "gray-matter";
+import issueParser from "issue-parser";
+const { test: hasFrontMatter } = matter;
 
 import { ReleasePackage } from "../releaseGroups.js";
 import { Repository } from "./git.js";
 
 export const DEFAULT_CHANGESET_PATH = ".changeset";
 
+/**
+ * The section name used for changesets that do not match any defined sections.
+ */
+export const UNKNOWN_SECTION = "_unknown";
+
+/**
+ * Additional metadata that can be used inside a changeset. This metadata should be set in a second front matter
+ * section.
+ * @example
+ *
+ * ```markdown
+ * ---
+ * "package-a": minor
+ * ---
+ * ---
+ * section: fix
+ * ---
+ *
+ * Changeset summary.
+ *
+ * Changeset details
+ * ```
+ */
+export interface FluidCustomChangesetMetadata {
+	/**
+	 * The section in release notes in which this changeset should be included. If a value is not provided, the changeset
+	 * is considered part of the "unknown section".
+	 */
+	section?: ReleaseNotesSectionName;
+
+	/**
+	 * If false, the changeset will not be included in release notes.
+	 *
+	 * @defaultValue `true`
+	 */
+	includeInReleaseNotes?: boolean;
+
+	/**
+	 * If true, the changeset will be ordered before the other changes in that section. If multiple changesets are
+	 * highlighted they will be internally sorted by date.
+	 *
+	 * @defaultValue `false`
+	 */
+	highlight?: boolean;
+}
+
+export const fluidCustomChangeSetMetadataDefaults: FluidCustomChangesetMetadata = {
+	section: undefined,
+	includeInReleaseNotes: true,
```

Comment:
I don't see a default value applied to `highlight` anywhere. The behavior might not change with this but it seems good to make it clear.
```suggestion
	section: undefined,
	includeInReleaseNotes: true,
	highlight: false,
```

Reply:
Fixed and protected against future mistakes of omitting defaults when new metadata is added.